content/api/commands/session.md

```diff
@@ -0,0 +1,729 @@
+  cy.session([username, password], () => {
+    cy.request('POST', `/api/login`, {
+      username,
+      password,
+    }).then((response) => {
+      expect(response.status).to.eq(200)
+    })
+  })
+})
+
+it('is a redundant test', () => {
+  /* which you can now delete! */
+})
+```
+
+### Cross-domain sessions
+
+It's possible to switch domains while caching sessions, just be sure to
+explicitly visit the domain in your login command before calling `cy.session()`.
+
+```jsx
+const login = (name) => {
+  if (location.hostname !== 'example.com') {
+    cy.visit('example.com')
+  }
+  cy.session(name, () => {
+    cy.visit('/login')
+    // etc
+  }, {
+    validate() {
+      cy.request('/whoami', {
+        headers: { 'Authorization' : localStorage.token }
+        method: 'POST'
+      }).its('statusCode').should('equal', '200')
+    }
+  })
+}
+
+it('t1', ()=>{
+  login('bob')
+  // do things on example.com
+})
+
+it('t2', ()=>{
+  cy.visit('anotherexample.com')
+  // do things on anotherexample.com
+})
+
+it('t3', ()=>{
+  login('bob')
+  // do things on example.com
+})
+```
+
+## Notes
+
+### When the page and active session data are cleared
+
+The page is cleared and all active session data (cookies, `localStorage`, and
+`sessionStorage`) across all domains are cleared automatically when
+`cy.session()` runs. This guarantees consistent behavior whether a session is
+being created or restored and allows you to switch sessions without first having
+to explicitly log out.
+
+|                      |              Current page cleared               |           Active session data cleared           |
+| -------------------- | :---------------------------------------------: | :---------------------------------------------: |
+| Before `setup`       | <Icon name="check-circle" color="green"></Icon> | <Icon name="check-circle" color="green"></Icon> |
+| Before `validate`    | <Icon name="check-circle" color="green"></Icon> |                                                 |
+| After `cy.session()` | <Icon name="check-circle" color="green"></Icon> |                                                 |
+
+Because calling `cy.session()` clears the current page in addition to restoring
+cached session data, [`cy.visit()`](/api/commands/visit) must always be
+explicitly called afterwards to ensure a page is visited.
+
+### Session caching
+
+Once created, a session for a given `id` is cached for the duration of the spec
+file. You can't modify a stored session after it has been cached, but you can
+always create a new session with a different `id`.
+
+In order to reduce development time, when running the Test Runner in "open"
+mode, sessions will be cached _between spec file runs_ as long as the `setup`
+function hasn't changed.
+
+### Explicitly clearing sessions
+
+When running the Test Runner in "open" mode, you can explicitly clear all
+sessions and re-run the spec file by clicking the "Clear All Sessions" button in
```

Comment:
Would be great to have an image of this as I'd have to search for where this  is just reading it from nothing.

Reply:
> * I feel like I'm missing the  overall 'why' of using cy.session. If you read the document from start to finish, it never seems to really answer this question fully. I start to get an idea on why I might use this when the examples start. It'd be great to just be really upfront after the first sentence of the page on why you want to use this. - This is great for testing logins, it will save you time, you won't have to manually juggle cookies, session data, etc... Maybe there's a blog post to point to, not  sure.

Does this (draft) blog post help explain it? https://cypress-io.ghost.io/blog/p/32959945-4e57-40d7-8a47-8e1ba67784cf/ I didn't want to get too much into the why in the API docs page, because I was concerned that it would make using the page as a reference more difficult, but instead focused on the how.

Reply:
> Would be great to have an image of this as I'd have to search for where this is just reading it from nothing.

I have the image in the [Instrument Panel](https://deploy-preview-4004--cypress-docs.netlify.app/api/commands/session#The-Instrument-Panel) section of the docs, which is linked to in the section you've called out. Should I duplicate the image there as well?

Reply:
Yah, probably just duplicate the image. Since I was reading straight through or if you're linked here directly, you may not know to scroll down. I must have missed the link.

Reply:
Image added in fb1161f46bbce5e921860ed6c6b70ba991c02576.